rfc/rfc-92/rfc-92.md

```diff
@@ -0,0 +1,231 @@
+
+The main implementation step here is to create abstraction called TableFormatPlugin which handles table format operations such as 
+
+
+1. Committing writes 
+2. Update and read table format specific Metadata operations 
+3. Timeline 
+4. Conflict Resolution
+5. Lock Provider
+6. Rollbacks
+
+
+The Hudi platform is responsible for managing the data path and can be configured with the table format plugin by default using Hudi native table format. Other table formats will have their own implementation of this abstraction.  
+
+### Commit Protocol:
+
+Hudi uses the creation of an action-complete file (e:g .commit, .deltacommit, .clean, ..)  in timeline folder to advertise the completion of an action on the table. With the TableFormatPlugin, this will become a two step process:
+
+1. Creation of the action complete file in .hoodie timeline 
+2. Storing the action completion time in the table format's commit metadata.
+
+The Hudi timeline is still very much used for all internal operations and the table format's commit metadat will be an overlay on top of this.
+With this, the action is only completed when both the above steps are completed. The plugin provided timeline needs to fence the timeline ensuring the definition of complete stays consistent.  This ensures the snapshot isolation is maintained.
+
+There will be a hudi table-property "hudi.table.format.plugin" to identify the table format with default being "native". This allows consistency in plugin behaviors across all hudi writers.
+
+### Metadata:
+
+If a different table format is configured, Hudi's metadata operations are replaced with the table format's metadata. This is done by adding a new adapter implementation of HoodieTableMetadata.
+
+### Timeline: 
+
+The timeline implementation ensures the external table format's (table format that is plugged-in) metadata state (e:g commit status) is the source of truth for all operations. 
+
+### Conflict Resolution
+
+Hudi already provides configurable support to plugin different strategies for identifying if two concurrent operations are conflicting at the file level. The 
+external pluggable formats may not have such fine-level granularity. The external pluggable formats needs to implement the strategy based on its metadata.
+
+### Locking
+Hudi has pluggable lock provider supports. External formats will need to rely on catalog or other means for providing locking semantics. The External table format implementation needs to provide custom lock provider to adapt their locking mechanisms to Hudi's LockProvider interface so that all write and table-service operations can concurrently operate. 
+
+### Rollbacks
+External Table format implementation  needs to rollback failed writes and also be able to also restore to previous commits. The external plugin implementation needs to keep enough states to be able to rollback.
+ 
+### Layout:
+
+The metadata corresponding to the table format (iceberg, ..) will be stored under .hoodie/ folder.
```

Comment:
This may be confusing, are we talking about the actual table format metadata or just the `lastSyncedInstant` ? If it's the former it might work for table formats which rely on absolute paths but may not work for ones which use relative paths.

Reply:
Good point. I was referring to the former. We can make this a configurable location in hoodie table properties.  Would need to ensure the location wont have any side-effects with hudi's partition path finding means.